pkg/logging/log_context_builder.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"fmt"
+)
+
+// LogContextBuilder allows building up log lines with context that is important
+// for debugging and tracing. This class helps create log line formatting
+// consistantly. Logging should always be in the form:
+// <Kind> "<Namespace>/<Name>": <message>
+type LogContextBuilder struct {
+	Kind      string
```

Comment:
can you make the type of this `Kind`, so that it can't be set to any arbitrary string?

Reply:
Oh nice idea!